File: backend/app/models/schemas/product.py
"""Product schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ProductCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    code: str = Field(..., min_length=1, max_length=120)
    team_id: int = Field(..., gt=0)
    description: str | None = None
    is_active: bool = True


class ProductUpdate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    code: str = Field(..., min_length=1, max_length=120)
    team_id: int = Field(..., gt=0)
    description: str | None = None
    is_active: bool = True


class ProductResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    team_id: int
    team_name: str | None = None
    code: str
    name: str
    description: str | None = None
    is_active: bool
    project_count: int = 0
    created_at: datetime
    updated_at: datetime

File: frontend/src/shared/types/product.ts
export interface ProductSummary {
  id: number;
  team_id: number;
  team_name: string | null;
  code: string;
  name: string;
  description: string | null;
  is_active: boolean;
  project_count: number;
  created_at: string;
  updated_at: string;
}

export interface ProductListResponse {
  items: ProductSummary[];
  total: number;
  page: number;
  page_size: number;
}

export interface ProductUpsertPayload {
  team_id: number;
  code: string;
  name: string;
  description?: string | null;
  is_active: boolean;
}
